src/components/Components/ComponentDetails/tabs/ComponentDetails.tsx

```diff
@@ -36,6 +42,20 @@ const ComponentDetails: React.FC<React.PropsWithChildren<ComponentDetailsProps>>
   const latestImageURL = results?.find((result) => result.name === RESULT_NAME);
   const componentImageURL = latestImageURL?.value ?? getLastestImage(component);
 
+  // Check if image controller is enabled for this cluster
+  const { isImageControllerEnabled } = useIsImageControllerEnabled();
+
+  // Check if user has permission to update image repository using RBAC
+  const [canUpdateImageRepository] = useAccessReviewForModel(ImageRepositoryModel, 'update');
+
+  // Only fetch ImageRepository if both image controller is enabled AND user has update permission
+  const shouldFetchOrShowImageRepository = isImageControllerEnabled && canUpdateImageRepository;
+  const [imageRepository, imageRepoLoaded, imageRepoError] = useImageRepository(
+    shouldFetchOrShowImageRepository ? component?.metadata?.namespace : null,
+    shouldFetchOrShowImageRepository ? component?.metadata?.name : null,
+    false,
+  );
```

Comment:
move both permission and IR hooks to the `ComponentImageRepositoryVisibility` component.
Also, If user doesn't have sufficient permission. Disable the edit option with permission tooltip.

Reply:
Updated. The new patch provide 'list' and 'edit' permission check and related tool-tips.